src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/StorePal.Windows.Export.cs

```diff
@@ -92,55 +90,54 @@ public void MoveTo(X509Certificate2Collection collection)
 
         public byte[] ExportPkcs12(Pkcs12ExportPbeParameters exportParameters, SafePasswordHandle password)
         {
-            bool tryAes256sha256 = exportParameters is
-                Pkcs12ExportPbeParameters.Default or
-                Pkcs12ExportPbeParameters.Pbes2Aes256Sha256;
-
-            byte[] exported = ExportPkcs12(tryAes256sha256, password, out bool aes256Sha256);
-
-            // What we asked for and what we got are the same - so return it as-is.
-            if (tryAes256sha256 == aes256Sha256)
-            {
-                return exported;
-            }
-
-            if (!tryAes256sha256)
-            {
-                Debug.Fail("3DES PKCS12 export failed.");
-                throw new CryptographicException();
-            }
-
-            return ReEncryptAndSealPkcs12(exported, password, Helpers.WindowsAesPbe);
+            return ExportPkcs12Core(exportParameters, password);
         }
 
         public byte[] ExportPkcs12(PbeParameters exportParameters, SafePasswordHandle password)
         {
-            byte[] exported = ExportPkcs12(preferAes256Sha256: true, password, out _);
+            byte[] exported = ExportPkcs12Core(null, password);
```

Comment:
Since we are going to re-encode it anyway, just let it export as whatever so we don't double re-encode it.